src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -225,10 +226,19 @@ def validate_resolve_keyvault(namespace):
 
 
 def validate_feature(namespace):
-    if namespace.feature:
-        invalid_pattern = re.compile(r'[^a-zA-Z0-9._-]')
-        invalid = re.search(invalid_pattern, namespace.feature)
-        if invalid:
-            raise CLIError("Feature name is invalid. Only alphanumeric characters, '.', '-' and '_' are allowed.")
-    else:
-        raise CLIError("Feature name cannot be empty.")
+    if namespace.feature is not None:
+        if '%' in namespace.feature:
+            raise CLIError("Feature name cannot contain the '%' character.")
+        if not namespace.feature:
+            raise CLIError("Feature name cannot be empty.")
```

Comment:
Could you please use the specific error type `InvalidArgumentValueError` instead of `CLIError`

Reply:
Is this a new error type? Do we need to change it in other existing commands too?

Reply:
Updated all new errors in _validators.py to `InvalidArgumentValueError`